src/components/ruleOverview/ruleOverview.js

```diff
@@ -17,15 +17,15 @@ class RuleOverviewFlyout extends Component {
         };
     }
 
-    onToggleStatus = () => {
-        this.setState({enabled: !this.state.enabled});
-    };
-
     onSave = () => {
         let promises = [];
         let updatedRules = [];
         this.props.rules.forEach(rule => {
-            if (rule.Enabled !== this.state.enabled) {
+            if (rule.Enabled === this.state.enabled) {
+                rule.Enabled = !this.state.enabled;
+                updatedRules.push(rule);
```

Comment:
The rule is pushed into the array and the promise is pushed into its array regardless of the if condition. Move these values out of the if and do them all the time.